### PR TITLE
Presentation: Learning snippets cleanup

### PR DESCRIPTION
**Pull request by grigasp**

### CODE REVIEW

**Review by roluk — CHANGES_REQUESTED** [review]
Please also delete reference to a non-existent `Backend.Initialization.RpcInterface` snippet from `PresentationRpcImpl.ts`.

Also, something needs to be done with snippet located at https://www.itwinjs.org/presentation/advanced/testing/

The code comments there refers to a VSTS item, has an ESLint disable comment, and a test that is skipped.

**Comment by grigasp** [comment]
> Please also delete reference to a non-existent `Backend.Initialization.RpcInterface` snippet from `PresentationRpcImpl.ts`.
> 
> Also, something needs to be done with snippet located at https://www.itwinjs.org/presentation/advanced/testing/
> 
> The code there refers to a VSTS item, has an ESLint disable comment, and a test that is skipped.

Cleaned up `PresentationRpcImpl.ts` docs, reworked `RulesetTesting` tests and docs. Please have a look.

File: docs/presentation/Advanced/Testing.md [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
 
-The package delivers an API that allows creating hierarchies for supplied
-iModels and rulesets. Consumers can then verify the result using tools of
-their liking. Our recommendation is to use snapshot testing for 2 reasons:
+The package delivers an API that allows creating hierarchies for supplied iModels and rulesets. Consumers can then verify the result using tools of their liking. Our recommendation is to use snapshot testing for 2 reasons:
 
-1. resulting hierarchies get rather large - testing the them in
-code might be difficult
-2. snapshots protect against regressions
+1. Resulting hierarchies get rather large - testing the them in code might be difficult.
```
Comment (roluk) [inline docs/presentation/Advanced/Testing.md:8]:
```suggestion
1. Resulting hierarchies get rather large — testing them in code might be difficult.
```

File: docs/presentation/Advanced/Testing.md [hunk 2]
```diff
@@ -1,33 +1,51 @@
...
+      [[include:Presentation.Testing.Rulesets.Content]]
+      ```
+
+3. Close iModel and terminate testing library:
+
+    ```ts
+    [[include:Presentation.Testing.Rulesets.Terminate]]
+    ```
+
+**Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. The path returned by `createSnapshotPath` function used in the tests should match the project structure and is outside the scope of this example.
```
Comment (roluk) [inline docs/presentation/Advanced/Testing.md:41]: > The path returned by `createSnapshotPath` function used in the tests should match the project structure [..]

Does chai-jest-snapshot really care where you place snapshot files? I think you should omit this sentence.

**Review by roluk — CHANGES_REQUESTED** [review]

File: docs/presentation/Advanced/Testing.md (hunk 2 above)
Comment (grigasp) [reply docs/presentation/Advanced/Testing.md:41]: I'm using this function in the example code and now including its content. I think it makes sense to tell what it's for.

Comment (roluk) [reply docs/presentation/Advanced/Testing.md:41]: Does chai-jest-snapshot really care where you place snapshot files though? I find this sentence potentially misleading.

Comment (roluk) [reply docs/presentation/Advanced/Testing.md:41]: Then let's not mislead people that a particular file structure is required for snapshots to work... What `createSnapshotPath` does is obvious from the name, but if you insist on over explaining it, you can do it without imposing our internal requirements to the end users:

```suggestion
**Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. `createSnapshotPath` function retrieves a path to the snapshot for the current test.
```

**Review by roluk — APPROVED** [review]

**Pull request merged**